libcudacxx/test/libcxx/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+
+set(LIBCXX_LIT_VARIANT "libcxx" CACHE STRING
+    "Configuration variant to use for LIT.")
+
+set(LIBCXX_TEST_LINKER_FLAGS "" CACHE STRING
+    "Additonal linker flags to pass when compiling the tests")
+
+# <nv/target> is found in the parent project, it is required in all headers now
+set(LIBCXX_TEST_COMPILER_FLAGS
+    "-I${libcudacxx_SOURCE_DIR}/include" CACHE STRING
+    "Additonal compiler flags to pass when compiling the tests")
+
+# sccache cannot handle the -Fd option generationg pdb files
+if (MSVC)
+  set(CMAKE_MSVC_DEBUG_INFORMATION_FORMAT Embedded)
+endif()
+
+# Ensure we catch all warnings
+if (MSVC)
+  set(LIBCUDACXX_TEST_COMPILER_FLAGS "${LIBCUDACXX_TEST_COMPILER_FLAGS} /W4 /WX")
+else()
+  set(LIBCUDACXX_TEST_COMPILER_FLAGS "${LIBCUDACXX_TEST_COMPILER_FLAGS} -Wall -Wextra")
+endif()
+
+# Intel OneAPI compiler has fast math enabled by default which breaks almost all floating point tests
+if (${CMAKE_CXX_COMPILER_ID} STREQUAL "IntelLLVM" OR ${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
+  set(LIBCUDACXX_TEST_COMPILER_FLAGS "${LIBCUDACXX_TEST_COMPILER_FLAGS} -fno-fast-math")
+endif()
+
+include(AddLLVM)
+
+set(LIBCUDACXX_BINARY_DIR "${CMAKE_CURRENT_BINARY_DIR}")
+set(LIBCUDACXX_TARGET_INFO "libcxx.test.target_info.LocalTI" CACHE STRING
+    "TargetInfo to use when setting up test environment.")
+set(LIBCUDACXX_EXECUTOR "None" CACHE STRING
+    "Executor to use when running tests.")
+set(LIBCUDACXX_TEST_TIMEOUT "200" CACHE STRING "Enable test timeouts (Default = 200, Off = 0)")
+set(AUTO_GEN_COMMENT "## Autogenerated by libcudacxx configuration.\n# Do not edit!")
+
+set(lit_site_cfg_path "${CMAKE_CURRENT_BINARY_DIR}/lit.site.cfg")
+configure_lit_site_cfg("${CMAKE_CURRENT_SOURCE_DIR}/lit.site.cfg.in" "${lit_site_cfg_path}")
+add_lit_testsuite(check-libcxx "Running libcu++ host only tests" "${CMAKE_CURRENT_BINARY_DIR}")
+
+find_program(libcudacxx_LIT lit REQUIRED)
+set(libcudacxx_LIT_FLAGS "" CACHE STRING "Semi-colon separated list of flags passed to the invocation of lit.")
```

Comment:
A lot of these option names are already used by the libcudacxx tests. Rename to `libcxx`?